contracts/contracts/harvest/OETHBaseHarvester.sol

```diff
@@ -0,0 +1,196 @@
+
+        // Collect all AERO
+        amoStrategy.collectRewardTokens();
+
+        uint256 aeroBalance = aero.balanceOf(address(this));
+        if (aeroBalance == 0) {
+            // Do nothing if there's no AERO to transfer
+            return;
+        }
+
+        // Transfer everything to Strategist
+        aero.safeTransfer(strategistAddr, aeroBalance);
+    }
+
+    /**
+     * @notice Harvests AERO from AMO strategy and then swaps some (or all)
+     *          of it into WETH to distribute yield and fee.
+     *         When `feeBps` is set to 10000 (100%), all WETH received is
+     *          sent to strategist.
+     *
+     * @param aeroToSwap Amount of AERO to swap
+     * @param minWETHExpected Min. amount of WETH to expect
+     * @param feeBps Performance fee bps (Sent to strategist)
+     * @param yieldRecipient Yield recipient (must be Vault or Dripper)
+     */
+    function harvestAndSwap(
+        uint256 aeroToSwap,
+        uint256 minWETHExpected,
+        uint256 feeBps,
+        address yieldRecipient
+    ) external onlyGovernorOrStrategist {
+        address strategistAddr = vault.strategistAddr();
+        require(strategistAddr != address(0), "Guardian address not set");
+
+        require(feeBps <= 10000, "Invalid Fee Bps");
+
+        // Collect all AERO
+        amoStrategy.collectRewardTokens();
+
+        uint256 aeroBalance = aero.balanceOf(address(this));
+        if (aeroBalance == 0) {
+            // Do nothing if there's no AERO to transfer/swap
+            return;
+        }
+
+        if (aeroToSwap > 0) {
+            require(aeroBalance >= aeroToSwap, "Insufficient balance for swap");
+            _doSwap(aeroToSwap, minWETHExpected);
+
+            // Figure out AERO left in contract after swap
+            aeroBalance = aero.balanceOf(address(this));
+        }
+
+        // Transfer out any leftover AERO after swap
+        if (aeroBalance > 0) {
+            aero.safeTransfer(strategistAddr, aeroBalance);
+        }
+
+        // Computes using all balance the contract holds,
+        // not just the WETH received from swap. Use `transferToken`
+        // if there's any WETH left that needs to be taken out
+        uint256 availableWETHBalance = weth.balanceOf(address(this));
+        // Computation rounds in favor of protocol
+        uint256 fee = (availableWETHBalance * feeBps) / 10000;
+        uint256 yield = availableWETHBalance - fee;
+
+        // Transfer yield to yield recipient if any
+        if (yield > 0) {
+            // Yields can only be sent to the Vault or the Dripper.
+            // There's no address(0) check since Vault will break if there's
+            // no Dripper address set.
+            require(
+                yieldRecipient == address(vault) ||
+                    yieldRecipient == vault.dripper(),
+                "Invalid yield recipient"
+            );
+            weth.safeTransfer(yieldRecipient, yield);
```

Comment:
This is really nit-picking and contains a lot of "if", but still. 

If the `strategist` make a mistake and pass `address(0)` for `yieldRecipient` and if the dripper is not enable (set to `address(0)`), then the yield can be sent to `address(0)`.

Maybe it could be worth to check that `yieldRecipient` is not `address(0)`? 

_I understand that this is a very unlikely to happen._

Reply:
So these checks are here to ensure that yields are sent only to the Vault or the Dripper and to no other address by accident. Dripper address is read from the vault.   
    
So there shouldn't be a case where either the Vault address or Dripper is set to address(0). If Vault is set to zero address, the calls to read Dripper and strategist address would fail. If Dripper is set to zero address, our rebase won't even work and would break a lot of things. 

The only way I see this breaking is if we upgrade the Vault to not revert when Dripper is a zero address and set it. I don't feel strongly about this, but happy to accommodate the change if you see it fit

Reply:
> If Dripper is set to zero address, our rebase won't even work and would break a lot of things.

I missed that part, so indeed it is safe 👍

Reply:
AFAIK the Vault rebases just fine when the dripper is set to zero address. I think it is a great check to add that the yieldRecipient is not a zero address

Reply:
I just checked, yeah, I was wrong. It's the mint that fails not rebase. Will add that check then

Reply:
Should be fixed now